src/dsp_tools/models/xmlvalue.py

```diff
@@ -23,14 +24,82 @@ def __init__(
+            self.value = KnoraStandoffXml(xmlstr_cleaned)
             self.resrefs = list({x.split(":")[1] for x in self.value.get_all_iris() or []})
+        elif val_type == "text" and node.get("encoding") == "utf8":
+            str_orig = "".join(node.itertext())
+            str_cleaned = self._cleanup_unformatted_text(str_orig)
+            self.value = str_cleaned
         elif val_type == "list":
             listname = cast(str, listname)
             self.value = listname + ":" + "".join(node.itertext())
         else:
             self.value = "".join(node.itertext())
+
+    def _cleanup_formatted_text(self, xmlstr_orig: str) -> str:
+        """
+        In a xml-encoded text value from the XML file,
+        there may be non-text characters that must be removed.
+        This method:
+            - removes the <text> tags
+            - replaces (multiple) line breaks by a space
+            - replaces multiple spaces or tabstops by a single space (except within <code> tags)
+
+        Args:
+            xmlstr_orig: original string from the XML file
+
+        Returns:
+            purged string, suitable to be sent to DSP-API
+        """
+        # remove the <text> tags
+        xmlstr = regex.sub("<text.*?>", "", xmlstr_orig)
+        xmlstr = regex.sub("</text>", "", xmlstr)
+
+        # replace (multiple) line breaks by a space
+        xmlstr = regex.sub("\n+", " ", xmlstr)
+
+        # replace multiple spaces or tabstops by a single space (except within <code> tags)
+        # the regex selects all spaces/tabstops not followed by </code> without <code in between.
+        # credits: https://stackoverflow.com/a/46937770/14414188
+        xmlstr = regex.sub("( {2,}|\t+)(?!(.(?!<code))*</code>)", " ", xmlstr)
+
+        # remove spaces after <br/> tags (except within <code> tags)
+        xmlstr = regex.sub("((?<=<br/?>) )(?!(.(?!<code))*</code>)", "", xmlstr)
+
+        # remove leading and trailing spaces
+        xmlstr = xmlstr.strip()
+
+        return xmlstr
+
+    def _cleanup_unformatted_text(self, string_orig: str) -> str:
+        """
+        In a utf8-encoded text value from the XML file,
+        there may be non-text characters that must be removed.
+        This method:
+            - removes the <text> tags
+            - replaces multiple spaces or tabstops by a single space
+            - replaces 3+ line breaks by 2 line breaks (=max. 1 empty line)
+
+        Args:
+            string_orig: original string from the XML file
+
+        Returns:
+            purged string, suitable to be sent to DSP-API
+        """
+        # remove the <text> tags
+        string = regex.sub("<text.*?>", "", string_orig)
+        string = regex.sub("</text>", "", string)
+
+        # replace multiple spaces or tabstops by a single space
+        string = regex.sub(r" {2,}|\t+", " ", string)
+
+        # replace 3+ line breaks by 2 line breaks
+        string = regex.sub(r"(\n ?){3,}", "\n\n", string)
```

Comment:
```suggestion
        string = regex.sub(r"(\n *){3,}", "\n\n", string)
```
I think you need this to be sure to match double newlines followed by multiple spaces